src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -119,30 +129,35 @@ def accept_bid(self, bid: Bid, energy: float = None,
             changed_bid = Bid(
                 str(uuid.uuid4()), residual_price, residual_energy,
                 buyer, seller,
-                original_bid_price=energy_portion * orig_price
+                original_bid_price=(1 - energy_portion) * orig_price
             )
+
             self.bids[changed_bid.id] = changed_bid
             self.bid_history.append(changed_bid)
 
             self._notify_listeners(MarketEvent.BID_CHANGED,
                                    existing_bid=bid, new_bid=changed_bid)
             residual = changed_bid
-
-            final_price = self._update_bid_fee_and_calculate_final_price(
-                energy, trade_rate, energy_portion, orig_price
-            ) if already_tracked is False else energy * trade_rate
-            bid = Bid(bid.id, energy * trade_rate, energy, buyer, seller,
+            revenue, fees, trade_price = calculate_trade_price_and_fees(
+                trade_offer_info, self.transfer_fee_ratio
+            )
+            self.market_fee += fees
+            final_price = energy * trade_price
+            bid = Bid(bid.id, final_price, energy, buyer, seller,
                       original_bid_price=energy_portion * orig_price)
         else:
-            final_price = self._update_bid_fee_and_calculate_final_price(
-                energy, trade_rate, 1, orig_price
-            ) if already_tracked is False else energy * trade_rate
-
-            bid = bid._replace(price=energy * trade_rate)
+            revenue, fees, trade_price = calculate_trade_price_and_fees(
+                trade_offer_info, self.transfer_fee_ratio
+            )
+            self.market_fee += fees
+            final_price = energy * trade_price
```

Comment:
Sorry, here I ym confused over the naming again. I followed the calculation of the trade_price and see that is indeed a price, not a rate. Why are you multiplying with the energy here? Or am I wrong and it is really a rate and just needs to renamed?

Reply:
Sorry, it really is a rate here. Will rename this